weighing.py

```diff
@@ -392,10 +432,126 @@ def create_quality_test(self):
     def draft(cls, weighings):
         pass
 
+    def get_invoice(self, party):
+        pool = Pool()
+        Journal = pool.get('account.journal')
+        Invoice = pool.get('account.invoice')
+        Company = pool.get('company.company')
+        context = Transaction().context
+
+        invoices = Invoice.search([
+            ('party', '=', party),
+            ('state', '=', 'draft'),
+            ('type', '=', 'in')])
+        if invoices:
+            invoice = invoices[0]
+        if not invoices:
+            journals = Journal.search([
+                ('type', '=', 'expense'),
+                ], limit=1)
+            if journals:
+                journal, = journals
+            else:
+                journal = None
+
+            invoice = Invoice()
+            invoice.company = Company(context['company'])
+            invoice.type = 'in'
+            invoice.journal = journal
+            invoice.party = party
+            invoice.invoice_address = party.address_get(type='invoice')
```

Comment:
why not invoice.on_change_party()  instead to search any field by yourself.

Reply:
We no longer need the function "get_invoice", we only create invoice lines.